satpy/composites/__init__.py

```diff
@@ -787,10 +785,20 @@ def __init__(self, name, common_channel_mask=True, **kwargs):
         Args:
             common_channel_mask (bool): If True, mask all the channels with
                 a mask that combines all the invalid areas of the given data.
+
         """
         self.common_channel_mask = common_channel_mask
         super(GenericCompositor, self).__init__(name, **kwargs)
 
+    @classmethod
+    def infer_mode(cls, data_arr):
+        """Guess at the mode for a particular DataArray."""
+        if 'mode' in data_arr.attrs['mode']:
```

Comment:
Maybe this should be `if 'mode' in data_arr.attrs:`?

Reply:
Yes, I've fixed it locally when I continued working on #902. Thanks.